storage/src/mmr/journaled.rs

```diff
@@ -124,20 +191,46 @@ impl<B: Blob, E: RStorage<B> + Metrics, H: Hasher> Mmr<B, E, H> {
 
     /// Sync any new elements to disk.
     pub async fn sync(&mut self) -> Result<(), Error> {
+        // Write peaks to metadata without clearing out old ones to ensure we can recover to a
+        // previous valid state if the later writes to the journal fail.
+        let peak_iterator = self.mem_mmr.peak_iterator();
+        for (peak_pos, _) in peak_iterator {
+            let digest = self.mem_mmr.get_node(peak_pos).await?.unwrap();
+            self.metadata
+                .put(U64::new(peak_pos), Bytes::copy_from_slice(digest.as_ref()));
+        }
+        self.metadata.sync().await.map_err(Error::MetadataError)?;
```

Comment:
I had originally suspected you'd append/sync journal before metadata (as you can unwind journal based an out-of-date metadata but can't unwind metadata if journal is behind).

Is there some rationale I'm missing?

Reply:
Hmm, no I think you're right -- this is a leftover from  my initial implementation of pruning where if you didn't do a full sync() before pruning, you'd have to write peaks first to ensure they weren't pruned before you had the change to write them.  This got messy though (see the TODO in prune) and I ended up just deciding to always do a full sync() up front before pruning to simplify things.  So I can probably flip the order now.

Reply:
ok, changed to write journal first, tests still pass and it's simpler since we can always clear the old peaks. Glad you caught that!